image/image_test.go

```diff
@@ -4166,3 +4173,64 @@ func (s *imageSuite) TestLocalSnapRevisionMatchingStoreRevision(c *C) {
 		},
 	})
 }
+
+func (s *imageSuite) TestSetupSeedLocalSnapWithInvalidArchitecture(c *C) {
+	restore := image.MockTrusted(s.StoreSigning.Trusted)
+	defer restore()
+
+	rootdir := filepath.Join(c.MkDir(), "image")
+	a64Snap := snaptest.MakeTestSnapWithFiles(c, arm64Snap, nil)
+
+	opts := &image.Options{
```

Comment:
Would it make sense to have a test here that uses image.Options.Architecture ? AFAICT this codepath is not tested(?) - so that it checks that with that override image building works?

Reply:
Definitely a good idea

Reply:
Added a unit test